core/src/main/java/org/elasticsearch/node/InternalSettingsPreparer.java

```diff
@@ -44,6 +47,9 @@
 
 public class InternalSettingsPreparer {
 
+    private static Logger logger = ESLoggerFactory.getLogger("settings");
+    private static DeprecationLogger deprecationLogger = new DeprecationLogger(logger);
```

Comment:
See my [suggestion](https://github.com/elastic/elasticsearch/pull/24059#discussion_r111127455) for how I think that we can avoid this.